docs/DeveloperGuide.md

```diff
@@ -1,38 +1,164 @@
+
+We aim to help fellow gym enthusiasts and new gym members to help keep
+track of their future and present training.
+=======
+### UI component
+
+###### [Back to table of contents](#table-of-contents)
+
+### Start component
+
+###### [Back to table of contents](#table-of-contents)
+
+### Add component
+
+The add mechanism is facilitated by `AddCommand`. 
+It extends `Command` and modifies the execute function to add an exercise.
+
+<img src="images/AddExerciseDiagram.png" width="450" />
+
+Given below is an example usage scenario and how the add mechanism behaves at each step.
+
+Step 1. The user enters the add command with the necessary arguments.
+
+Step 2. The input is processed by the `Parser` to separate out the arguments and creates the exercise to be added, 
+`toAdd`.
+
+Step 3. The `AddCommand` calls `execute()` which calls `WorkList#getCurrentWorkout()` to return `currentWorkout`.
+
+Step 4. Finally, `addExercise()` is called and `toAdd` is added to `currentWorkout`.
+
+###### [Back to table of contents](#table-of-contents)
+
+### List component
 
-## Product scope
-### Target user profile
+###### [Back to table of contents](#table-of-contents)
 
-{Describe the target user profile}
+### View component
+The View component is facilitated by `Parser`,`Ui`,`WorkoutList`,`Command` and `ViewCommand`, where the user will 
+enter a specific workout date and the number of exercises on that date will be displayed
 
-### Value proposition
+Below are the specific steps on how to use the view function and how the mechanism will flow:
+
+* Step 1: We will assume that the user has started a workout on two specific dates, `11/02/22` which was added with the following command `/start 11/02/22`
+  and `12/02/22`, which was added with the following command `/start 12/02/22`.
+* Step 2: The user will then use the following command `/view 11/02/22` will be taken into the parser
+  and will return a list of exercises done on that specified date.
+
+<img src="images/ViewDiagram.png" width="450" />
+
+###### [Back to table of contents](#table-of-contents)
+
+### Delete component
+The deletion mechanism is facilitated by 'Parser', 'DeleteCommand', 'Workout', 'WorkoutList' and 'UI', where a Workout object will be deleted according to the command inputted by the user and removed from the workout list.
+
+<img src="images/DeleteWorkoutDiagram.png" width="450" />
+
+Below is an example usage scenario and how the deletion mechanism behaves at each step:
+
+Step 1: Assume that the user has already added a workout on 21/03/23 into the WorkoutList using the following command, /start 21/03/23
+
+Step 2: The user input of /delete 21/03/23 will be taken in for the parser and an object of class DeleteCommand will be returned.
+
+Step 3: The execute method in the DeleteCommand class that is overrides will be called with parameter date and will iterate through workoutList looking for a workout that matches. It will then remove the workout from the workoutList.
```

Comment:
Perhaps it will be better for commands to be in markdown code snippets format. It may also be better to highlight class name so to make it more easily identified.